src/avtk/clipselector.cxx

```diff
@@ -269,13 +299,25 @@ int ClipSelector::handle(int event)
 					{ "Save" },
 					{ "Special"},
 					{ "Beats",  0,   0, 0, FL_SUBMENU | FL_MENU_DIVIDER },
-					{"1       "},
-					{"2"},
-					{"4"},
-					{"8"},
-					{"16"},
-					{"32"},
-					{"64"},
+					{"1 ", 0, setLengthCb, (void*)1,0},
+					{"2 ", 0, setLengthCb, (void*)2,0},
+					{"4 ", 0, setLengthCb, (void*)4,0},
+					{"8 ", 0, setLengthCb, (void*)8,0},
+					{"16 ", 0, setLengthCb, (void*)16,0},
+					{"32 ", 0, setLengthCb, (void*)32,0},
+					{"64 ", 0, setLengthCb, (void*)64,0},
+					{0},
+					{ "Bars to record",  0,   0, 0, FL_SUBMENU | FL_MENU_DIVIDER},
+					{"1 ", 0, setRecordBarsCb, (void*)1,0},
+					{"2 ", 0, setRecordBarsCb, (void*)2,0},
+					{"3 ", 0, setRecordBarsCb, (void*)3,0},
+					{"4 ", 0, setRecordBarsCb, (void*)4,0},
+					{"5 ", 0, setRecordBarsCb, (void*)5,0},
+					{"6 ", 0, setRecordBarsCb, (void*)6,0},
+					{"7 ", 0, setRecordBarsCb, (void*)7,0},
+					{"8 ", 0, setRecordBarsCb, (void*)8,0},
+					{"Custom", 0, setRecordBarsCb, (void*)-2,0},
```

Comment:
I'm not sure how I feel about the (void*)-2, and -1. What purpose do Custom and Endless have?